### ISSUE TITLE
TF2 keras model.fit() VS model.evaluate print different loss and AUC

### ISSUE THREAD

**Issue by jasonxcx** [tensorflow/tensorflow#60177]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.11

### Custom Code

Yes

### Python version

3.7.3

### Current Behaviour?

```shell
My model generate 3 binary outputs with MMoE module, and I define a custom metric to combine these three AUC in callbacks. As I am using model.fit to train my dataset,  it prints the final training AUC in the last epoch is 0.98. While I am using model.evaluate() on the SAME training dataset, it turns out the AUC is only 0.65. Can anyone can help me with it? Btw, I do not use BatchNormalization and Dropout layers, and also the batch_size is the same for both fitting and evaluation process.

I used exactly same dataset and same batch_size, I really don’t know why it occurs?
```

### Standalone code to reproduce the issue

```shell
class MergeMetrics(tf.keras.callbacks.Callback):
    def __init__(self,**kargs):
        super(MergeMetrics,self).__init__(**kargs)

    def on_epoch_begin(self,epoch, logs={}):
        return

    def on_epoch_end(self, epoch, logs={}):
        logs['merge_auc'] = 0.7 *logs["output_1_auc"]+ 0.2*logs["output_2_auc"] + 0.1*logs["output_3_auc"]


model = Modelname(some_parameters)
model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate, beta_1, beta_2),
            loss={
                "output_1_auc": ‘binary_crossentropy’
                "output_2_auc": ‘binary_crossentropy’
                "output_3_auc": ‘binary_crossentropy’
            },
            loss_weights=[1,1,1],
            metrics={
                "output_1_auc": [tf.keras.metrics.AUC(name='auc')],
                "output_2_auc": [tf.keras.metrics.AUC(name='auc')],
                "output_3_auc": [tf.keras.metrics.AUC(name='auc')]})


checkpoint = MergeMetrics()
checkpoint_filepath = os.path.join(path,'model')
model_check = tf.keras.callbacks.ModelCheckpoint(
        filepath = checkpoint_filepath,
        # filepath = path,
        monitor= "merge_auc",
        save_best_only = True,
        mode='max',
        save_weights_only = True,
        save_freq="epoch")

train_dataset = tf.data.Dataset.from_tensor_slices((train_data,                                           (y_train.output_1, y_train.output_2, y_train.output_3))).shuffle(10*batch_size).batch(batch_size)

model.fit(train_dataset, epochs=5, callbacks=[checkpoint,model_check])
model.load_weights(checkpoint_filepath) ## load weights of the best epoch
auc  = model.evaluate(train_data, [y_train.output_1, y_train.output_2, y_train.output_3], batch_size=batch_size)
```

### Relevant log output

```shell
RESULT:
Fit process : output_1_auc:0.9862,  output_2_auc: 0.9665, output_3_auc:0.5014, merge_auc: 0.9338
Evaluation process: output_1_auc: 0.7124500870704651, output_2_auc:0.6924731135368347, output_3_auc: 0.6774155497550964
```
</details>

**Comment by jasonxcx** [comment]
Hi @synandi 
Thank you so much, but unluckily it still not work, my training result is 80+%, while my evaluation result is 60+%

**Comment by jasonxcx** [comment]
neg labels and pos label are clustered, and the shuffle() seems not working
